modules/gltf/gltf_document.cpp

```diff
@@ -5323,11 +5316,21 @@ void GLTFDocument::_convert_csg_shape_to_gltf(CSGShape3D *p_current, GLTFNodeInd
 	Ref<ImporterMesh> mesh;
 	mesh.instantiate();
 	{
-		Ref<Mesh> csg_mesh = csg->get_meshes()[1];
+		Ref<ArrayMesh> csg_mesh = csg->get_meshes()[1];
 
 		for (int32_t surface_i = 0; surface_i < csg_mesh->get_surface_count(); surface_i++) {
 			Array array = csg_mesh->surface_get_arrays(surface_i);
-			Ref<Material> mat = csg_mesh->surface_get_material(surface_i);
+
+			Ref<Material> mat;
+
+			Ref<Material> mat_surface_override = csg_mesh->surface_get_material(surface_i);
+			if (mat_surface_override.is_valid()) {
+				mat = mat_surface_override;
+			}
+			Ref<Material> mat_override = csg->get_material_override();
```

Comment:
Should this be inside of `} else {`? Or which should take priority? I think that whichever should take priority should go first and the lesser priority should be placed inside `} else {`.

Reply:
1. highest priority (matching mesh instance) is geometric node material
2. second highest is the per mesh instances

We do this in reverse order.

Reply:
To be honest I have no preference, I can rewrite it your way for ease of reading.